zerver/data_import/mattermost.py

```diff
@@ -433,46 +477,67 @@ def process_batch(lst: List[Dict[str, Any]]) -> None:
                        subscriber_map: Dict[int, Set[int]],
                        output_dir: str,
                        masking_content: bool,
                        stream_id_mapper: IdMapper,
+                       huddle_id_mapper: IdMapper,
                        user_id_mapper: IdMapper,
                        user_handler: UserHandler,
                        username_to_user: Dict[str, Dict[str, Any]],
                        zerver_realmemoji: List[Dict[str, Any]],
                        total_reactions: List[Dict[str, Any]]) -> None:
-
-    stream_id_to_recipient_id = {
-        d['type_id']: d['id']
-        for d in zerver_recipient
-        if d['type'] == Recipient.STREAM
-    }
-
-    def get_stream_recipient_id(raw_message: ZerverFieldsT) -> int:
-        receiver_id = raw_message['receiver_id']
-        stream_id = stream_id_mapper.get(receiver_id)
-        recipient_id = stream_id_to_recipient_id[stream_id]
+    recipient_id_to_recipient_type = {}
+    stream_id_to_recipient_id = {}
+    huddle_id_to_recipient_id = {}
+    user_id_to_recipient_id = {}
+
+    for d in zerver_recipient:
+        recipient_id_to_recipient_type[d['id']] = d['type']
+        if d['type'] == Recipient.STREAM:
+            stream_id_to_recipient_id[d['type_id']] = d['id']
+        elif d['type'] == Recipient.HUDDLE:
+            huddle_id_to_recipient_id[d['type_id']] = d['id']
+        if d['type'] == Recipient.PERSONAL:
+            user_id_to_recipient_id[d['type_id']] = d['id']
+
+    def get_recipient_id_from_receiver_name(receiver_name: str, recipient_type: int) -> int:
+        if recipient_type == Recipient.STREAM:
+            receiver_id = stream_id_mapper.get(receiver_name)
+            recipient_id = stream_id_to_recipient_id[receiver_id]
+        elif recipient_type == Recipient.HUDDLE:
+            receiver_id = huddle_id_mapper.get(receiver_name)
+            recipient_id = huddle_id_to_recipient_id[receiver_id]
+        elif recipient_type == Recipient.PERSONAL:
+            receiver_id = user_id_mapper.get(receiver_name)
+            recipient_id = user_id_to_recipient_id[receiver_id]
         return recipient_id
 
-    process_posts(
-        num_teams=num_teams,
-        team_name=team_name,
-        realm_id=realm_id,
-        post_data=post_data,
-        get_recipient_id=get_stream_recipient_id,
-        subscriber_map=subscriber_map,
-        output_dir=output_dir,
-        is_pm_data=False,
-        masking_content=masking_content,
-        user_id_mapper=user_id_mapper,
-        user_handler=user_handler,
-        username_to_user=username_to_user,
-        zerver_realmemoji=zerver_realmemoji,
-        total_reactions=total_reactions,
-    )
+    if num_teams == 1:
+        post_types = ["channel_post", "direct_post"]
+    else:
+        post_types = ["channel_post"]
+        print("Skipping importing huddles and PMs since there are multiple teams in the export")
```

Comment:
Should this be a `logging.warning()` instead?

Reply:
updated.